### ISSUE TITLE
Overlapping --uidmap arguments could yield a better error message

### ISSUE THREAD

**Issue by eriksjolund** [containers/podman#13149]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Overlapping __--uidmap__ arguments could yield a more informative error message.
Podman should be able to catch this by itself before starting `crun`.

**Steps to reproduce the issue:**

1.

```
[root@laptop ~]# useradd test40
[root@laptop ~]# machinectl shell test40@
Connected to the local host. Press ^] three times within 1s to exit session.
[test40@laptop ~]$ podman run --rm -ti --uidmap 0:0:1000 --uidmap 0:0:1000 docker.io/library/alpine
Trying to pull docker.io/library/alpine:latest...
Getting image source signatures
Copying blob 59bf1c3509f3 done  
Copying config c059bfaa84 done  
Writing manifest to image destination
Storing signatures
Error: OCI runtime error: crun: writing file `/proc/27849/gid_map`: Invalid argument
[test40@laptop ~]$ 
```

**Describe the results you received:**

```
Error: OCI runtime error: crun: writing file `/proc/27849/gid_map`: Invalid argument
```

**Describe the results you expected:**

```
Error: overlapping UID mapping
```
or for instance

```
Error: overlapping UID mapping passed with --uidmap
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 4.0.0-rc2
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.0-dev
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc35.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 8
  distribution:
    distribution: fedora
    variant: workstation
    version: "35"
  eventLogger: journald
  hostname: laptop
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1035
      size: 1
    - container_id: 1
      host_id: 2638224
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1035
      size: 1
    - container_id: 1
      host_id: 2703760
      size: 65536
  kernel: 5.16.5-200.fc35.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 28530917376
  memTotal: 33452257280
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.4.2-1.fc35.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.2
      commit: f6fbc8f840df1a414f31a60953ae514fa497c748
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1035/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc35.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 9124700160
  swapTotal: 9124700160
  uptime: 6h 47m 29.92s (Approximately 0.25 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/test40/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/test40/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/1035/containers
  volumePath: /home/test40/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.0-rc2
  Built: 1642797034
  BuiltTime: Fri Jan 21 21:30:34 2022
  GitCommit: ""
  GoVersion: go1.16.13
  OsArch: linux/amd64
  Version: 4.0.0-rc2


```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-4.0.0-0.2.rc2.fc35.x86_64
```

**Comment by giuseppe** [comment]
duplicate of: https://github.com/containers/storage/issues/1127
